src/bundle.ts

```diff
@@ -95,3 +108,52 @@ export class BuildBundler extends Transform {
         });
   }
 }
+
+export type ResolveFileCallback = (a: string) => void;
+export type RejectFileCallback = (err: Error) => void;
+export type DeferredFileCallbacks = {
+  resolve: ResolveFileCallback; reject: RejectFileCallback;
+};
+
+class StreamLoader implements UrlLoader {
+  bundler: BuildBundler;
+
+  // Store files that have not yet entered the Analyzer stream here.
+  // Later, when the file is seen, the DeferredFileCallback can be
+  // called with the file contents to resolve its loading.
+  deferredFiles = new Map<string, DeferredFileCallbacks>();
+
+  constructor(bundler: BuildBundler) {
+    this.bundler = bundler;
+  }
+
+  canLoad(url: string): boolean {
+    return this.bundler.analyzer.analyzer.canResolveUrl(url);
+  }
+
+  getFile(url: string): File {
+    return this.bundler.files.get(url);
+  }
+
+  load(url: string): Promise<string> {
```

Comment:
I'd like to propose a convention that all Promise-returning functions should be async.

Not only do you get `await` goodness, but you can also be confident that any thrown error becomes a rejection. This is more predictable than splitting the failure codepath, with some becoming rejections and some being thrown up the call stack.